### PR TITLE
[#9158] switch conch to using sign/verify methods

### PR DESCRIPTION
**Pull request by reaperhulk**
I'm not sure what to put for the news fragment, and the twisted project will need to decide if they want to actually bump their dep or potentially just allow deprecation warnings from cryptography.

refs https://github.com/pyca/cryptography/issues/3659

### CODE REVIEW

**Review by alex — APPROVED** [review]

**Comment by alex** [comment]
Code looks good. Please add a newsfile as well.

File: src/twisted/conch/newsfragments/9158.misc [hunk 1]
```diff
@@ -0,0 +1 @@
+Bumped minimum cryptography version to 1.5.
```
Comment (mithrandi) [inline src/twisted/conch/newsfragments/9158.misc]: The contents of .misc files is ignored, so usually they are left empty.

**Comment by glyph** [comment]
As for the news fragment I think this can be a `.misc` (it's "just" keeping up with the deprecation cycle of another library?) or a brief `.bugfix` note just noting the raised minimum version.

**Comment by reaperhulk** [comment]
@glyph I have it as a misc now, so if that's good enough 👍

**Pull request merged**